### PR TITLE
[FIX] libc++: header tests

### PR DESCRIPTION
**Pull request by eseiler**

### CODE REVIEW

File: include/seqan3/core/debug_stream/debug_stream_type.hpp [hunk 1]
```diff
@@ -12,7 +12,7 @@
 
 #pragma once
 
-#include <iosfwd>
+#include <ostream>
```
Comment (smehringer) [inline include/seqan3/core/debug_stream/debug_stream_type.hpp:15]: This now counter-acts the purpose of a lightweight debug_stream_type?

Comment (eseiler) [reply include/seqan3/core/debug_stream/debug_stream_type.hpp:15]: Kind of
I can do different includes for libstdc++ (gcc) and libc++ (llvm) :D

Comment (smehringer) [reply include/seqan3/core/debug_stream/debug_stream_type.hpp:15]: there is no iosfw equivalent for libc++?

I would vote for different includes then :)

Comment (eseiler) [reply include/seqan3/core/debug_stream/debug_stream_type.hpp:15]: I'll have to check what the issue was, but we are accessing stuff that requires a defined `std::ostream`. Whatever we need is apparently in `<iosfwd>` for libstdc++, but not in libc++

Comment (eseiler) [reply include/seqan3/core/debug_stream/debug_stream_type.hpp]: Actually, we don't need `<ostream>` here, but in `pretty_printing.hpp`. 

So, good catch :+1:

**Review by smehringer — APPROVED** [review]

**Pull request merged**